Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-46.ca-central-1.compute.internal
 Git:  git version 2.9.5
Sha: efa09506097255e072122d20c0a6563a09aee71b
Containerized:False
Number of Error(s): 5
src-react/components/Queries.tsx(132,52): error TS2554: Expected 1 arguments, but got 2.
src-react/components/Queries.tsx(237,48): error TS2554: Expected 1 arguments, but got 2.
make[2]: *** [esp/src/CMakeFiles/eclwatch.dir/build.make:312: esp/src/build/dist/stub.eclwatch.js] Error 1
make[1]: *** [CMakeFiles/Makefile2:10418: esp/src/CMakeFiles/eclwatch.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2021-08-05_12-21-44
time logs started
Prep time   :     9 sec (00:00:09)
Build time  :   165 sec (00:02:45)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   174 sec (00:02:54)
time logs end

[Build error:](https://gist.github.com/e0e526799f57297a47e7894c05fcaf07)
Build log [Build log](https://gist.github.com/HPCCSmoketest/e0e526799f57297a47e7894c05fcaf07#file-result-21-08-05-12-18-36-log)

--- a/esp/src/src-react/components/Queries.tsx
+++ b/esp/src/src-react/components/Queries.tsx
@@ -14,6 +14,7 @@ import { DojoGrid, selector } from "./DojoGrid";
 
 const FilterFields: Fields = {
     "QueryID": { type: "string", label: nlsHPCC.ID, placeholder: nlsHPCC.QueryIDPlaceholder },
+    "Priority": { type: "queries-priority", label: nlsHPCC.Priority },
     "QueryName": { type: "string", label: nlsHPCC.Name, placeholder: nlsHPCC.QueryNamePlaceholder },
     "PublishedBy": { type: "string", label: nlsHPCC.PublishedBy, placeholder: nlsHPCC.PublishedBy },
     "WUID": { type: "string", label: nlsHPCC.WUID, placeholder: "W20130222-171723" },
@@ -24,19 +25,13 @@ const FilterFields: Fields = {
     "Activated": { type: "queries-active-state", label: nlsHPCC.Activated }
 };
 
-function formatQuery(filter: any, wuid?: string) {
-    if (wuid !== undefined) {
-        return {
-            WUID: wuid
-        };
-    }
-    const retVal = { ...filter };
-    if (filter.StartDate) {
-        retVal.StartDate = new Date(filter.StartDate).toISOString();
-    }
-    if (filter.EndDate) {
-        retVal.EndDate = new Date(filter.StartDate).toISOString();
-    }
+function formatQuery(filter: any) {
+    const retVal = {
+        ...filter,
+        PriorityLow: filter.Priority,
+        PriorityHigh: filter.Priority
+    };
+    delete retVal.Priority;
     return retVal;
 }
 
@@ -70,7 +65,7 @@ export const Queries: React.FunctionComponent<QueriesProps> = ({
 
     //  Grid ---
     const gridStore = useConst(store || ESPQuery.CreateQueryStore({}));
-    const gridQuery = useConst(formatQuery(filter, wuid));
+    const gridQuery = useConst(formatQuery(filter));
     const gridSort = useConst([{ attribute: "Id" }]);
     const gridColumns = useConst({
         col1: selector({
@@ -136,6 +131,13 @@ export const Queries: React.FunctionComponent<QueriesProps> = ({
                 return `<a href='#/queries/${row.QuerySetId}/${Id}' class='dgrid-row-url'>${Id}</a>`;
             }
         },
+        priority: {
+            label: nlsHPCC.Priority,
+            width: 80,
+            formatter: function (priority, row) {
+                return priority === undefined ? "" : priority;
+            }
+        },
         Name: {
             label: nlsHPCC.Name
         },
@@ -168,11 +170,11 @@ export const Queries: React.FunctionComponent<QueriesProps> = ({
     });
 
     const refreshTable = React.useCallback((clearSelection = false) => {
-        grid?.set("query", formatQuery(filter, wuid));
+        grid?.set("query", formatQuery(filter));
         if (clearSelection) {
             grid?.clearSelection();
         }
-    }, [filter, grid, wuid]);
+    }, [filter, grid]);
 
     //  Command Bar  ---
     const buttons = React.useMemo((): ICommandBarItemProps[] => [

--- a/esp/src/src-react/components/forms/Fields.tsx
+++ b/esp/src/src-react/components/forms/Fields.tsx
@@ -45,7 +45,7 @@ const Dropdown: React.FunctionComponent<DropdownProps> = ({
 export type FieldType = "string" | "number" | "checkbox" | "datetime" | "dropdown" | "link" | "links" | "progress" |
     "workunit-state" |
     "file-type" | "file-sortby" |
-    "queries-suspend-state" | "queries-active-state" |
+    "queries-priority" | "queries-suspend-state" | "queries-active-state" |
     "target-cluster" | "target-dropzone" | "target-server" | "target-group" |
     "target-dfuqueue" |
     "logicalfile-type" | "dfuworkunit-state";
@@ -101,6 +101,11 @@ interface FileTypeField extends BaseField {
 
 interface FileSortByField extends BaseField {
     type: "file-sortby";
+    value?: string;
+}
+
+interface QueriesPriorityField extends BaseField {
+    type: "queries-priority";
     value?: string;
 }
 
@@ -170,7 +175,7 @@ interface ProgressField extends BaseField {
 type Field = StringField | NumericField | CheckboxField | DateTimeField | DropdownField | LinkField | LinksField | ProgressField |
     WorkunitStateField |
     FileTypeField | FileSortByField |
-    QueriesSuspendStateField | QueriesActiveStateField |
+    QueriesPriorityField | QueriesSuspendStateField | QueriesActiveStateField |
     TargetClusterField | TargetDropzoneField | TargetServerField | TargetGroupField |
     TargetDfuSprayQueueField |
     LogicalFileType | DFUWorkunitStateField;
@@ -589,6 +594,25 @@ export function createInputs(fields: Fields, onChange?: (id: string, newValue: a
                             { key: "Oldest", text: nlsHPCC.Oldest },
                             { key: "Smallest", text: nlsHPCC.Smallest },
                             { key: "Largest", text: nlsHPCC.Largest }
+                        ]}
+                        onChange={(ev, row) => onChange(fieldID, row.key)}
+                        placeholder={field.placeholder}
+                    />
+                });
+                break;
+            case "queries-priority":
+                field.value = field.value === undefined ? "" : field.value;
+                retVal.push({
+                    id: fieldID,
+                    label: field.label,
+                    field: <Dropdown
+                        key={fieldID}
+                        selectedKey={field.value.toString()}
+                        options={[
+                            { key: "", text: nlsHPCC.None },
+                            { key: "0", text: nlsHPCC.Low },
+                            { key: "1", text: nlsHPCC.High },
+                            { key: "2", text: nlsHPCC.SLA }
                         ]}
                         onChange={(ev, row) => onChange(fieldID, row.key)}
                         placeholder={field.placeholder}

